### DISCUSSION TITLE
Welcome to SourceSpec Discussions!

### DISCUSSION THREAD

**Discussion by claudiodsf** [SeismicSource/sourcespec#13] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by anniejerk** [comment]
Hi,
I’d like to try using a different velocity model instead of iasp91 to compute the takeoff angles when incorporating the radiation pattern and focal mechanism in the analysis. Is this currently possible?

**Reply by claudiodsf** [reply]
Hi,

what kind of model would you like to use? A different global model (e.g., AK135) or a local/regional one?

In the first case, that is currently not possible (but easy to implement), in the second case, you can use a NonLinLoc grid through the `NLL_time_dir` configuration parameter.

**Reply by anniejerk** [reply]
Thanks for the quick reply! I would like to use a 1D regional model

**Reply by claudiodsf** [reply]
Ok, so you have to go the NonLinLoc route! 

Take a look at the `vel2grid` and `grid2time` utils in the NonLinLoc package. Doc is here: http://alomax.free.fr/nlloc/

**Reply by anniejerk** [reply]
perfect! Thank you so much

**Reply by claudiodsf** [reply]
Let me also ping @krisvanneste, since he was working on a different strategy to compute takeoff angles from regional models (see https://github.com/SeismicSource/sourcespec/issues/57). Maybe he's got something to share...

**Reply by krisvanneste** [reply]
Hi,

Indeed I'm using a local 1D velocity model to compute travel times and takeoff angles, which I then write to the trace headers before launching SourceSpec.

I forgot that I started this issue last year. I think it should be possible to incorporate the class I use to calculate travel times and takeoff angles in sourcespec, so that it can be used if a velocity model is given in the configuration file, but I think I was waiting for some feedback before embarking on it.

**Reply by claudiodsf** [reply]
Ok, so Kris's solution is not for right now 😉. We'll work on it in the near future 💪

**Reply by anniejerk** [reply]
Thank you for all help :D

**Reply by anniejerk** [reply]
@claudiodsf  I've been working with NonLinLoc and have generated some grids, but since this is my first time using it, I'm having a bit of trouble getting your code to run smoothly with these grids.

When I attempt to run it, I encounter the following error messages:


input must be a scalar  
NS.BER.00.HHE: Unable to get P arrival time: skipping trace  
input must be a scalar  
NS.BER.00.HHN: Unable to ...
The arrivals seemed to work previously, as they are stored in a separate .xml file, so I'm not quite sure what's going wrong now.

Would you happen to have any suggestions for what might be causing this? Also, is there a way to increase the verbosity level of the output so I can get a better idea of where the issue is occurring?

Thanks so much in advance for your help—I really appreciate your time and expertise!

**Reply by claudiodsf** [reply]
Hi @anniejerk. I don't have enough clues to help you. Could you try sharing somewhere a complete working example, with input files and your configuration?

**Reply by anniejerk** [reply]
[sourcespec (3).zip](https://github.com/user-attachments/files/20369696/sourcespec.3.zip)

**Reply by claudiodsf** [reply]
Thanks,

I could run your test and found two problems:

1. A bug in SouceSpec preventing the grids to being correctly read. This is fixed in the latest development version, which you can install through `pip install git+https://github.com/SeismicSource/sourcespec.git`

2. Your grids are not correctly constructed: the epicenter is on the (0, 0) position, which is the lower-left corner of the grid. Since your model is 1D, you could construct a simpler 2D grid with something like `VGGRID 2 250 10 0.0 0.0 -5.0 10.0 10.0 10.0 SLOW_LEN` (note the first value of "2" for the number of x points). You should then use `GTMODE GRID2D ANGLES_YES`

**Reply by anniejerk** [reply]
Thank you so much!
